.devcontainer/docker-compose.yml

```diff
@@ -0,0 +1,60 @@
+version: '3.4'
+
+services:
+
+  rabbitmq:
+    image: rabbitmq:3.8.3-management
+    environment:
+        RABBITMQ_DEFAULT_USER: guest
+        RABBITMQ_DEFAULT_PASS: guest
+    ports:
+      - '5672:5672'
+      - '15672:15672'
+
+    healthcheck:
+      test: rabbitmq-diagnostics -q ping
+      interval: 30s
+      timeout: 30s
+      retries: 5
+    networks:
+      - aiida
+
+  postgres:
+    image: postgres:12
+    ports:
+      - '5432:5432'
+    networks:
+      - aiida
+    environment:
+      POSTGRES_HOST_AUTH_METHOD: trust
+    # volumes:
+    #   - db-volume:/var/lib/postgresql/data
+
+  aiida:
+    #image: "aiidateam/aiida-core:main"
+    image: "aiida-core-dev"
+    build:
+      # need to add the parent directory to context to copy over new configure-aiida.sh
+      context: ..
+      dockerfile: .devcontainer/Dockerfile
+    user: aiida
+    environment:
+      DB_HOST: postgres
+      BROKER_HOST: rabbitmq
+
+    # no need for /sbin/my_init
+    entrypoint: tail -f /dev/null
+    volumes:
+      - ..:/home/aiida/aiida-core:cached
+    #   - db-volume:/var/lib/assets
+    networks:
+      - aiida
+    depends_on:
+      - rabbitmq
+      - postgres
+
+# volumes:
+#   db-volume:
```

Comment:
Do this need to be kept here? Since it is useful for having data stored persistent I suggest keeping it and adding more details.

Reply:
I can delete it.

This was meant as an indicator of how one could make the database persistent (e.g. if one wanted to use this as a production setup).

Reply:
in the end I decided to remove them as they are untested.

it's easy enough to reintroduce them and go to a setup with persistent data

Reply:
Thanks, but it is a bit confusing for me how this codespace is working, if I understand correctly, the container is opened in a remote machine provided by GitHub. How is this persistent volumes mapping can mapping to the local DB file or local file repository?

Reply:
Indeed this would have been intended for running the devcontainer locally through VS Code.

I am not aware of a way of persisting files after github codespaces deletes the containers.